src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -651,16 +642,28 @@ internal void ExecutePackageDownload(string name, PackageVersion version, string
                 }
 
                 // form header version pairs and download and install all packages
-                allPackageVersions
-                        .Select(x => new PackageDownloadHandle(x.Item1, x.Item2))
-                        .ToList()
-                        .ForEach(x => DownloadAndInstall(x, downloadPath));
+                dependencyVersionHeaders
+                    .Select((dep, i) => {
+                        // Note that Name will be empty when calling from DownloadAndInstallPackage.
+                        // This is currently not an issue, as the code path belongs to the Workspace Dependency
+                        // extension, which does not display dependencies names.
+                        var dependencyPackageHeader = version.full_dependency_ids[i];
+                        return new PackageDownloadHandle()
+                        {
+                            Id = dependencyPackageHeader._id,
+                            VersionName = dep.version,
+                            Name = dependencyPackageHeader.name
+                        };
+                    })
+                    .ToList()
+                    .ForEach(x => DownloadAndInstall(x, downloadPath));
             }
         }
 
         /// <summary>
         ///     Returns a newline delimited string representing the package name and version of the argument
         /// </summary>
+        [Obsolete("No longer used. Remove in 3.0.")]
```

Comment:
@mjkkirschner I thought we made it clear somewhere that these APIs are not part of Dynamo SDK? If not, let's do that..

Reply:
@QilongTang we only made that statement in new extension binaries like WorkspaceReferencesExtension - this is a public method in `DynamoCoreWPF` - we can't remove this method now without some of the other work on reorganizing or componentizing the API we've been discussing.

Reply:
Thank you for clarifying. I thought this is in WorkspaceReference project. My bad